File: judge-core/src/main/java/cn/edu/zjnu/acm/judge/core/ExecuteResult.java
package cn.edu.zjnu.acm.judge.core;

import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

/**
 * @author zhanhb
 */
@Builder(builderClassName = "Builder", toBuilder = true)
@Value
@SuppressWarnings("FinalClass")
public class ExecuteResult {

    private long time;
    private long memory;
    @NonNull
    private Status code;
    private int exitCode;
    private String message;

    public boolean isSuccess() {
        return code == Status.ACCEPTED;
    }

}

File: site/src/main/kotlin/cn/edu/zjnu/acm/judge/core/ExecuteResult.kt
package cn.edu.zjnu.acm.judge.core

/**
 * @author zhanhb
 */
data class ExecuteResult(
        val time: Long = 0,
        val memory: Long = 0,
        val code: Status,
        val exitCode: Int = 0,
        val message: String? = null
) {
    val isSuccess: Boolean
        get() = code === Status.ACCEPTED

}
